pkg/operation/botanist/resource_manager.go

```diff
@@ -101,3 +159,110 @@ func (b *Botanist) DeployGardenerResourceManager(ctx context.Context) error {
 func (b *Botanist) ScaleGardenerResourceManagerToOne(ctx context.Context) error {
 	return kubernetes.ScaleDeployment(ctx, b.K8sSeedClient.Client(), kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameGardenerResourceManager), 1)
 }
+
+func (b *Botanist) mustBootstrapGardenerResourceManager(ctx context.Context) (bool, error) {
+	shootAccessSecret := gutil.NewShootAccessSecret(resourcemanager.SecretNameShootAccess, b.Shoot.SeedNamespace)
+	if err := b.K8sSeedClient.Client().Get(ctx, client.ObjectKeyFromObject(shootAccessSecret.Secret), shootAccessSecret.Secret); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		return true, nil // Shoot access secret does not yet exist.
+	}
+
+	renewTime, err2 := time.Parse(time.RFC3339, shootAccessSecret.Secret.Annotations[resourcesv1alpha1.ServiceAccountTokenRenewTimestamp])
+	if err2 != nil {
+		return false, fmt.Errorf("could not parse renew timestamp: %w", err2)
```

Comment:
Nit (because of corner case): Isn't it better to handle this case gracefully (e.g. delete secret + start bootstrap), so that we can get out of this situation automatically?

Reply:
I wouldn't do it here since it's no "accepted" scenario for automatic re-bootstrapping. What I mean is that this timestamp cannot become unparseable because of a glitch in the system or whatever - the only way is that an operator manually tampers with it. I would rather want to show this error in this case.

Reply:
Fair enough 👍